integration/mediation-tests/tests-transport/src/test/resources/artifacts/ESB/server/repository/deployment/server/synapse-configs/default/proxy-services/MTOMMIMEBoundaryWhenContentTypePreservedTestBackendProxy.xml

```diff
@@ -0,0 +1,27 @@
+<!--
+  ~ Copyright (c)2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+  ~
+  ~ WSO2 Inc. licenses this file to you under the Apache License,
+  ~ Version 2.0 (the "License"); you may not use this file except
+  ~ in compliance with the License.
+  ~ You may obtain a copy of the License at
+  ~
+  ~    http://www.apache.org/licenses/LICENSE-2.0
+  ~
+  ~ Unless required by applicable law or agreed to in writing,
+  ~ software distributed under the License is distributed on an
+  ~ "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+  ~ KIND, either express or implied.  See the License for the
+  ~ specific language governing permissions and limitations
+  ~ under the License.
+  -->
+<proxy xmlns="http://ws.apache.org/ns/synapse" name="MTOMMIMEBoundaryWhenContentTypePreservedTestBackendProxy"
+       startOnLoad="true" trace="disable" transports="http https">
+    <description/>
+    <target>
+        <inSequence>
+            <log level="full"/>
```

Comment:
How does this ensure Mime boundaries are present in the request ?
Does this fails without the fix ? I doubt it because we are only checking for status code 200 in the assertion.

Reply:
When trying to build the message, if the MIME boundaries are wrong, message would not be built but an exception would be thrown. This is not for checking whether MIME boundary exists but to verify that existing MIME boundary in Content-Type matches with the actual MIME boundaries. Due to a regression issue introduced by a fix for preserving headers, Content-Type MIME boundary and the actual MIME boundaries mismatched and the back end was unable to build the message. This is to test that. 

Yes the test fails without the fix. I made it fail first.